WordPress/Classes/ViewRelated/Menus/MenuItemsViewController.m

```diff
@@ -0,0 +1,866 @@
+            parentItem = parentItem.parent;
+        }
+    }
+}
+
+- (MenuItemInsertionView *)addNewInsertionViewWithOrder:(MenuItemInsertionOrder)insertionOrder forItemView:(MenuItemView *)itemView
+{
+    NSInteger index = [self.stackView.arrangedSubviews indexOfObject:itemView];
+    MenuItemInsertionView *insertionView = [[MenuItemInsertionView alloc] init];
+    insertionView.delegate = self;
+    insertionView.insertionOrder = insertionOrder;
+    
+    switch (insertionOrder) {
+        case MenuItemInsertionOrderAbove:
+            insertionView.indentationLevel = itemView.indentationLevel;
+            break;
+        case MenuItemInsertionOrderBelow:
+            insertionView.indentationLevel = itemView.indentationLevel;
+            index++;
+            break;
+        case MenuItemInsertionOrderChild:
+            insertionView.indentationLevel = itemView.indentationLevel + 1;
+            index += 2;
+            break;
+    }
+    
+    NSLayoutConstraint *heightConstraint = [insertionView.heightAnchor constraintEqualToConstant:MenuItemsStackableViewDefaultHeight];
+    heightConstraint.priority = UILayoutPriorityDefaultHigh;
+    heightConstraint.active = YES;
+    
+    [self.insertionViews addObject:insertionView];
+    [self.stackView insertArrangedSubview:insertionView atIndex:index];
+    
+    [insertionView.widthAnchor constraintEqualToAnchor:self.stackView.widthAnchor].active = YES;
+    
+    return insertionView;
+}
+
+- (void)insertInsertionItemViewsAroundItemView:(MenuItemView *)toggledItemView
+{
+    if (self.isEditingForItemViewInsertion) {
+        [self removeItemInsertionViews:NO];
+    }
+    
+    self.isEditingForItemViewInsertion = YES;
+    
+    self.itemViewForInsertionToggling = toggledItemView;
+    toggledItemView.showsCancelButtonOption = YES;
+    toggledItemView.showsEditingButtonOptions = NO;
+    
+    [self addNewInsertionViewWithOrder:MenuItemInsertionOrderAbove forItemView:toggledItemView];
+    [self addNewInsertionViewWithOrder:MenuItemInsertionOrderBelow forItemView:toggledItemView];
+    [self addNewInsertionViewWithOrder:MenuItemInsertionOrderChild forItemView:toggledItemView];
+}
+
+- (void)insertItemInsertionViewsAroundItemView:(MenuItemView *)toggledItemView animated:(BOOL)animated
+{
+    BOOL wasEditing = self.isEditingForItemViewInsertion;
+
+    CGRect previousRect = toggledItemView.frame;
+    CGRect updatedRect = toggledItemView.frame;
+    
+    [self insertInsertionItemViewsAroundItemView:toggledItemView];
+    
+    if (!animated) {
+        return;
+    }
+    
+    // since we are adding content above the toggledItemView, the toggledItemView (focus) will move downwards with the updated content size
+    updatedRect.origin.y += MenuItemsStackableViewDefaultHeight;
+    
+    for (MenuItemInsertionView *insertionView in self.insertionViews) {
+        insertionView.hidden = YES;
+        insertionView.alpha = 0.0;
+    }
+    [UIView animateWithDuration:0.3 animations:^{
```

Comment:
You can replace the 0.3 with [WPAnimationDurationDefault](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/System/WPGUIConstants.m#L9)